version: "3"
services:
  pg:
    image: postgres:12.2
    environment:
      - POSTGRES_PASSWORD=password
      - POSTGRES_USER=postgres
      - POSTGRES_HOST=pg
      - POSTGRES_DB=contests
    expose:
      - 5432
    volumes:
      - pg_data:/var/lib/postgresql/data
      - ./database/test-pg-data.sql:/docker-entrypoint-initdb.d/test-pg-data.sql
    healthcheck:
      test: ["CMD", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5
  mongo:
    image: nemoload/nemoload/halan-task_mongo
    volumes:
      - node_modules:/app/node_modules
      - ./:/app
    expose:
      - 27017

  node:
    image: nemoload/halan-task_node
    environment:
      - POSTGRES_PASSWORD=password
      - POSTGRES_USER=postgres
      - POSTGRES_HOST=pg
      - POSTGRES_DB=contests
    expose:
      - 3000
    links:
      - pg
      - mongo
    depends_on:
      - pg
      - mongo
    volumes:
      - node_modules:/app/node_modules
    restart: on-failure
  nginx:
    image: nginx:latest
    links:
      - node
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf
      - ./nginx/server.crt:/etc/nginx/ssl/server.crt
      - ./nginx/server.key:/etc/nginx/ssl/server.key
    ports:
      - 80:80
      - 443:443
    depends_on:
      - node
    restart: always

volumes:
  node_modules:
  pg_data:
  mongo_data:
